### PR TITLE
Initial notification service to subscribe to SNS topic and send emails

### PR DESCRIPTION
**Pull request by dan-hill2802**

### CODE REVIEW

**Comment by hansnqyr** [comment]
looks good

File: lambda_mailer.tf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      LOG_LEVEL = "${var.log_level}"
+      REGION = "${var.region}"
+      MAX_THREADS  = "${var.max_threads}"
+    }
+  }
+  depends_on = ["aws_cloudwatch_log_group.sns_to_ses_mailer"]
+}
+
+resource "aws_iam_role" "lambda_sns_to_ses_mailer" {
+  name = "lambda_ucfs_sns_to_ses_mailer"
```
Comment (mattburgess) [inline lambda_mailer.tf]: Probably don't want 'ucfs' in a generic module's resource name?

File: lambda_mailer.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  policy_arn = "${aws_iam_policy.ses_send_mail.arn}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_sns_to_ses_mailer_access_s3" {
+  role       = "${aws_iam_role.lambda_sns_to_ses_mailer.name}"
+  policy_arn = "${aws_iam_policy.ses_send_mail_read_s3.arn}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_write_cloud_watch_logs" {
+  policy_arn = "${aws_iam_policy.write_cloud_watch_logs.arn}"
```
Comment (mattburgess) [inline lambda_mailer.tf]: Nit: role and policy_arn are the other way around on all your other policy attachments.

File: lambda_mailer.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  role = "${aws_iam_role.lambda_sns_to_ses_mailer.name}"
+}
+
+resource "aws_cloudwatch_log_group" "sns_to_ses_mailer" {
+  name              = "/aws/lambda/sns_to_ses_mailer"
+  retention_in_days = 180
```
Comment (mattburgess) [inline lambda_mailer.tf]: Do you want to make this configurable? Other people using this module may have different retention policies?

File: s3.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+data "aws_iam_policy_document" "ses_send_mail_read_s3" {
+  statement {
+    sid = "SendMailS3BucketReadAccess"
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+      "s3:GetObject",
+      "s3:GetObjectVersion",
+      "s3:GetObject",
```
Comment (mattburgess) [inline s3.tf]: This duplicates line 37

**Review by mattburgess — CHANGES_REQUESTED** [review]

File: lambda_mailer.tf (hunk 1 above)
Comment (dan-hill2802) [reply lambda_mailer.tf]: Good spot, not sure why it ever had that in the name

File: lambda_mailer.tf (hunk 2 above)
Comment (dan-hill2802) [reply lambda_mailer.tf]: The latter is the IDE plugin's default, so might change the others instead.

File: s3.tf (hunk 4 above)
Comment (dan-hill2802) [reply s3.tf]: Another good spot

File: lambda_mailer.tf (hunk 3 above)
Comment (dan-hill2802) [reply lambda_mailer.tf]: Can do. I imagine more being made configurable through the iterations.

**Review by mattburgess — APPROVED** [review]

**Pull request merged**